core/controllers/profile.py

```diff
@@ -247,6 +248,10 @@ def get(self):
         """Handles GET requests."""
         return_url = self.request.get('return_url', self.request.uri)
         # Validating return_url for no external redirections.
+        if re.search('create', return_url) is not None and (
+                user_services.has_fully_registered(self.user_id)):
+            self.redirect(python_utils.convert_to_bytes(return_url))
```

Comment:
@kevintab95, According to [this test](https://github.com/oppia/oppia/pull/6498/files#diff-67dc9c6c76f4d36708b88c2cbb62f56bR83) we are expecting this handler to redirect to creators page, right? If so can you please check why it's not working as expected?

[I think this might be a regression after fixing #6383, can you please investigate?]

Reply:
Hi @DubeySandeep, you're absolutely right! Nice catch. The original fix seems to be to avoid redirection to domains other than *.oppia.org but original regex (`re.match('^/[^//]', return_url)`) seemed to output True for most cases. [See this comment](https://github.com/oppia/oppia/pull/6413#issue-259441452).
![image](https://user-images.githubusercontent.com/11008603/66392448-889ca900-e9ed-11e9-9a25-3140a91930c9.png)

Also, it does not handle the case where the return_url belongs to the request domain but is a complete URL (eg. https://www.oppia.org/create/0).